apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -1149,6 +1123,27 @@ namespace MWMechanics
             break;
         }
 
+        case ESM::MagicEffect::CurePoison:
+            actor.getClass().getCreatureStats(actor).getActiveSpells().purgeEffect(ESM::MagicEffect::Poison);
+            break;
+        case ESM::MagicEffect::CureParalyzation:
+            actor.getClass().getCreatureStats(actor).getActiveSpells().purgeEffect(ESM::MagicEffect::Paralyze);
+            break;
+        case ESM::MagicEffect::CureCommonDisease:
+            actor.getClass().getCreatureStats(actor).getSpells().purgeCommonDisease();
+            break;
+        case ESM::MagicEffect::CureBlightDisease:
+            actor.getClass().getCreatureStats(actor).getSpells().purgeBlightDisease();
+            break;
+        case ESM::MagicEffect::CureCorprusDisease:
+            actor.getClass().getCreatureStats(actor).getSpells().purgeCorprusDisease();
+            break;
+        case ESM::MagicEffect::Dispel:
+            actor.getClass().getCreatureStats(actor).getActiveSpells().purgeAll(magnitude);
```

Comment:
Well, dispel seems to maybe be bugged in vanilla. I tested with some different magnitudes, and 50% seems to always succeed in removing all effects, while 40% and lower seem to never succeed. 45% seems sometimes succeed and sometimes not.

The magic effect description in the TES Construction Set is
"This effect removes magicka-based spell effects from the subject. It does not affect abilities, disease, curses, or constant magic item effects. The magnitude is the chance an effect is removed."
which makes it sound like maybe the % chance is supposed to be rolled separately for each effect, and with 45% I have been able to get a situation where some effects were removed while others weren't.